src/gateway/events/Message.ts

```diff
@@ -40,6 +42,17 @@ export async function Message(this: WebSocket, buffer: WS.Data) {
 
 	if (process.env.WS_VERBOSE)
 		console.log(`[Websocket] Incomming message: ${JSON.stringify(data)}`);
+	if (process.env.WS_DUMP) {
+		if (this.session_id) {
+			fs.mkdirSync(path.join("dump", this.session_id), { recursive: true });
+			fs.writeFileSync(path.join("dump", this.session_id, `${Date.now()}.in.json`), JSON.stringify(data, null, 2));
+		}
+		else {
+			fs.mkdirSync(path.join("dump", "unknown"), { recursive: true });
+			fs.writeFileSync(path.join("dump", "unknown", `${Date.now()}.in.json`), JSON.stringify(data, null, 2));
+			console.log("Unknown session id, dumping to unknown folder");
+		}
+	}
```

Comment:
Don't use sync fs calls here. Change the import to `fs/promises` and use async fs

```suggestion
	if (process.env.WS_DUMP) {
		if (this.session_id) {
			await fs.mkdir(path.join("dump", this.session_id), { recursive: true });
			await fs.writeFile(path.join("dump", this.session_id, `${Date.now()}.in.json`), JSON.stringify(data, null, 2));
		}
		else {
			await fs.mkdir(path.join("dump", "unknown"), { recursive: true });
			await fs.writeFile(path.join("dump", "unknown", `${Date.now()}.in.json`), JSON.stringify(data, null, 2));
			console.log("Unknown session id, dumping to unknown folder");
		}
	}
```